src/main/java/org/spoken_tutorial/health/elasticsearch/controllers/HomeController.java

```diff
@@ -0,0 +1,294 @@
+        }
+
+        else {
+            resultMap.put(queueId, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @PutMapping("/updateDocument/{documentId}/{documentType}/{documentPath}")
+    public Map<Long, String> updateDocument(@PathVariable String documentId, @PathVariable String documentType,
+            @PathVariable String documentPath, @RequestParam String category, @RequestParam String topic,
+            @RequestParam String outlinePath) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
+        Map<Long, String> resultMap = new HashMap<>();
+
+        if (queuemnt != null) {
+            if (doesFileExist(documentPath)) {
+                queuemnt.setDocumentPath(documentPath);
+                queuemnt.setMessage("success");
+                Optional category1 = Optional.ofNullable(category);
+                if (category1.isPresent()) {
+                    queuemnt.setCategory((String) category1.get());
+                }
+
+                Optional topic1 = Optional.ofNullable(topic);
+                if (topic1.isPresent()) {
+                    queuemnt.setTopic((String) topic1.get());
+                }
+
+                Optional outlinePath1 = Optional.ofNullable(outlinePath);
+                if (outlinePath1.isPresent()) {
+                    if (doesFileExist((String) outlinePath1.get())) {
+                        queuemnt.setOutlinePth((String) outlinePath1.get());
+                    }
+
+                }
+
+                queRepo.save(queuemnt);
+                resultMap.put(queuemnt.getQueueId(), "success");
+            }
+
+            else {
+                resultMap.put(queuemnt.getQueueId(), "failed");
+            }
+        } else {
+            resultMap.put(null, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @PutMapping("/updateDocumentRank/{documentId}/{documentType}/{rank}")
+    public Map<Long, String> updateDocumentRank(@PathVariable String documentId, @PathVariable String documentType,
+            @PathVariable int rank) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
+        Map<Long, String> resultMap = new HashMap<>();
+
+        if (queuemnt != null) {
+            queuemnt.setRank(rank);
+            queuemnt.setMessage("success");
+            resultMap.put(queuemnt.getQueueId(), "success");
+
+        } else {
+            resultMap.put(null, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @DeleteMapping("/deleteDocument/{documentType}/{documentTypeId}")
+    public Map<Long, String> deleteDocument(@PathVariable String documentId, @PathVariable String documentType) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
+        Map<Long, String> resultMap = new HashMap<>();
+
+        if (queuemnt != null) {
+
+            queRepo.delete(queuemnt);
```

Comment:
this should not delete from the database. it will just add a deleteDocument request to db.